### PR TITLE
565 feature geo inference integration

### LINKED ISSUE #565: Feature: geo-inference integration
**Is your feature request related to a problem? Please describe.**
I would like to integrate [geo-inference python package](https://pypi.org/project/geo-inference/) which facilitates the use of self-contained portable models to complete extractions from geospatial imagery.

**Describe the solution you'd like**
inference_segmentation module should fully support and use the geo-inference package.

**Describe alternatives you've considered**
None.

**Additional context (What problems does it solve?)**
- 2X inference speed-up with the use of batches, samplers and dataloaders.
- Resolves inference bugs #537 
- Reduces boilerplate code in inference_segmentation module; easy testing and code maintainability.
- Eliminates model architecture code with the use of TorchScript

### PR DESCRIPTION
**Pull request by valhassan**
Thank you for submitting your **PR** and helping us with **GDL**.
Please read the template below, fill it out as appropriate, and make additional changes to your code as needed. 
Please feel free to submit your PR even if it doesn't satisfy all of the requirements below.
If you need assistance or review from a maintainer, add the label __Status: Help Needed__ or __Status: Review Needed__ respectively.
After review, a maintainer will add the label __Status: Revision Needed__ if further work is required for the PR to be merged.

# Description
This PR fully impacts inference_segmentation module in geo-deep-learning. Refer to the issue for more details.
Fixes #565 

## Type of change
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other 

## How Has This Been Tested?
A shakedown test for inference_srgmentation has been added to this PR.

## Screenshots (if applicable):

# Checklist:
If you're unsure about any of these, don't hesitate to ask. We're here to help! 
- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code
- [x] My PR has a descriptive title
- [x] My code follows PEP8 (most of it)
- [x] My code have the proper documentation in the code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new errors
- [ ] My code is covered by the existing tests (if applicable)
- [x] My code includes tests that show that my feature works
- [ ] My PR passes Travis CI tests

_If your PR does not fulfill all of the requirements in the checklist above, that's OK!_ If you need help, @-mention a maintainer and/or add the __Status: Help Needed__ label.

### CODE REVIEW

**Review by CharlesAuthier — APPROVED** [review]

File: config/augmentation/basic_augmentation_segmentation.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ augmentation:
 
   # Augmentations done immediately before conversion to torch tensor
   normalization: # Normalization: parameters for finetuning. See examples below:
-    mean:  # -> mean: [0.485, 0.456, 0.406]
-    std:  # -> std: std: [0.229, 0.224, 0.225])
+    mean: [1e-8, 1e-8, 1e-8] # -> mean: [0.485, 0.456, 0.406]
+    std: [1e-8, 1e-8, 1e-8] # -> std: std: [0.229, 0.224, 0.225])
```
[the remark this answers is not in the document]
Comment (valhassan) [reply config/augmentation/basic_augmentation_segmentation.yaml:15]: These are just default values with no meaning; it is there because some tests rely on the default values. You're right, do not use for actual experiments

File: .github/workflows/github-actions-ci.yml [hunk 2]
```diff
@@ -1,60 +1,22 @@
...
+      run:
+        shell: bash -el {0}
     steps:
-      - name: Free Disk Space (Ubuntu)
-        uses: jlumbroso/free-disk-space@main
+      - name: checkout repository
+        uses: actions/checkout@v4.1.2
+      
+      - name: create environment
+        uses: conda-incubator/setup-miniconda@v3
```
Comment (ms888ekb) [inline .github/workflows/github-actions-ci.yml:14]: 👍

File: config/augmentation/basic_augmentation_segmentation.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ms888ekb) [reply config/augmentation/basic_augmentation_segmentation.yaml:15]: I believe, [0.485, 0.456, 0.406] is a set of mean values for the ImageNet dataset

File: evaluate_segmentation.py [hunk 3]
```diff
@@ -124,7 +124,9 @@ def main(params):
     gpkg_name_ = []
 
     for aoi in tqdm(list_aois, desc='Evaluating from input list', position=0, leave=True):
-        output_path = working_folder / f"{aoi.aoi_id}_pred.tif" if not inference_image else inference_image
+        print(f"\nEvaluating AOI: {aoi.aoi_id}")
```
Comment (ms888ekb) [inline evaluate_segmentation.py:127]: do we want to switch to logging,info() instead of print()?

File: config/augmentation/basic_augmentation_segmentation.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LucaRom) [reply config/augmentation/basic_augmentation_segmentation.yaml:15]: @valhassan I understand it is needed for testing, but aren't the transformation loaded by default in our training script? Maybe we could set stdevs values to 1 to be sure they have no impact if a user don't define them. For the mean I guess it would have almost no impact as it is.

@ms888ekb I think you are right, I found that information too.

[the remark this answers is not in the document]
Comment (ms888ekb) [reply config/augmentation/basic_augmentation_segmentation.yaml:15]: most of pretrained model available from torch use these values as default normalization (mean and std)
it's ok to change them as long as we don't rely on pretrained torch models

File: evaluate_segmentation.py (hunk 3 above)
Comment (valhassan) [reply evaluate_segmentation.py:127]: will take out the debug print statements. it's not meant to be there!

File: config/augmentation/basic_augmentation_segmentation.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (valhassan) [reply config/augmentation/basic_augmentation_segmentation.yaml:15]: Understood.

The new placeholder values will be 

mean: [0.0, 0.0, 0.0]
std: [1, 1, 1]

This should have no effect on the inputs

**Review by ms888ekb — APPROVED** [review]
